Sources/StreamChat/Database/DTOs/MessageDTO.swift

```diff
@@ -51,10 +51,15 @@ class MessageDTO: NSManagedObject {
         
         prepareDefaultSortKeyIfNeeded()
     }
-    
+
+    /// We use this value as the offset for the sorting key for unsent messages. Even if the local time of the device
+    /// is different from the server time, we want to be sure the newly sent message is always the last one. By applying
+    /// the same offset to all unsent message, we make sure the order of the messages is maintained.
+    static let localSortingKeyTimeOffset: TimeInterval = 24 * 60 * 60
+
     /// Makes sure the `defaultSortingKey` value is computed and set.
     fileprivate func prepareDefaultSortKeyIfNeeded() {
-        let newSortingKey = locallyCreatedAt ?? createdAt
+        let newSortingKey = locallyCreatedAt.map { $0.addingTimeInterval(Self.localSortingKeyTimeOffset) } ?? createdAt
```

Comment:
disclaimer: this is unrelated to your change and possible N/A

From a db perspective computing the sort key sounds is very inefficient (all rows/objects will get a computed value which will be used to sort, gets slower as you data set grows). You normally want your sort key to come from an indexed column

Reply:
The value is not computed on the fly. We add an additional column that's re-computed every time the object is changed (see the `willSave` method) and the db uses this value for sorting. This is a limitation of CoreData which doesn't allow sorting by more complex expression (see also channels where we sort by `max(createdAt, lastMessageAt)`.

However, this column is not indexed atm and it should be. Fixing 👍